Failing test: Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/expandable_flyout/alert_details_preview_panel_rule_preview·cy·ts - Alert details expandable flyout rule preview panel rule preview "before each" hook for "should display rule preview and its sub sections" "before each" hook for "should display rule preview and its sub sections"

A test failed on a tracked branch

```
AssertionError: Timed out retrying after 150000ms: Expected <span.euiLoadingSpinner.css-1fsqg4c-euiLoadingSpinner-l> not to exist in the DOM, but it was continuously found.

Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `Alert details expandable fl...`
    at waitForAlerts (webpack:///./tasks/alerts.ts:454:28)
    at waitForAlertsToPopulate (webpack:///./tasks/create_new_rule.ts:915:16)
    at Context.eval (webpack:///./e2e/investigations/alerts/expandable_flyout/alert_details_preview_panel_rule_preview.cy.ts:47:30)
```

First failure: [kibana-on-merge - main](https://buildkite.com/elastic/kibana-on-merge/builds/90580#019ce1b6-fedc-481a-b379-71aaa42f2cdf)

<!-- kibanaCiData = {"failed-test":{"test.class":"Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/expandable_flyout/alert_details_preview_panel_rule_preview·cy·ts","test.name":"Alert details expandable flyout rule preview panel rule preview \"before each\" hook for \"should display rule preview and its sub sections\" \"before each\" hook for \"should display rule preview and its sub sections\"","test.failCount":1}} -->

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/expandable_flyout/alert_details_preview_panel_rule_preview.cy.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/expandable_flyout/alert_details_preview_panel_rule_preview.cy.ts
@@ -45,6 +45,11 @@ describe(
       createRule(rule);
       visit(ALERTS_URL);
       waitForAlertsToPopulate();
+      // Wait for any in-flight alerts search that may have been triggered by a
+      // background rule evaluation immediately after waitForAlertsToPopulate's
+      // 500 ms idle window closed. A second idle check here ensures the spinner
+      // is gone before we try to expand the alert row (fixes #257389).
+      cy.waitForNetworkIdle('/internal/search/privateRuleRegistryAlertsSearchStrategy', 500);
       expandAlertAtIndexExpandableFlyout();
       clickRuleSummaryButton();
     });

